src/audio/crossover/crossover_generic.c

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Google LLC. All rights reserved.
+//
+// Author: Sebastiano Carlucci <scarlucci@google.com>
+
+#include <stdint.h>
+#include <sof/audio/format.h>
+#include <sof/audio/eq_iir/iir.h>
+#include <sof/audio/component.h>
+#include <sof/audio/format.h>
+#include <sof/audio/crossover/crossover.h>
+
+// static void crossover_generic(struct crossover_state *state,
+//                               int32_t x, int32_t y[], int32_t num_sinks)
+// {
+//         /* PASS THROUGH FOR NOW */
+//         for (int i = 0; i < num_sinks; i++) {
+//                 y[i] = x;
+//         }
+// }
+
+/*
+ * \brief Runs input in through the LR4 filter and returns it's output.
+ */
+static int32_t crossover_generic_process_lr4(int32_t in, lr4_state *lr4)
+{
+	int32_t z;
+
+	/* Cascade two biquads with same coefficients in series */
+	z = iir_process_biquad(in, lr4->coef, lr4->delay);
+	return iir_process_biquad(z, lr4->coef, &lr4->delay[2]);
```

Comment:
modularizing the processing for the biquads also helps in reusing the same coefficient array.